yt/visualization/_commons.py

```diff
@@ -13,28 +14,39 @@
 SUPPORTED_FORMATS = AGG_FORMATS + [".eps", ".ps", ".pdf", ".svg"]
 
 
-def validate_image_name(filename, suffix: str = ".png") -> str:
+def normalize_extension_string(s: str) -> str:
+    return f".{s.lstrip('.')}"
+
+
+def validate_image_name(filename, suffix: Optional[str] = None) -> str:
     """
     Build a valid image filename with a specified extension (default to png).
     The suffix parameter is ignored if the input filename has a valid extension already.
     Otherwise, suffix is appended to the filename, replacing any existing extension.
     """
-    # sanitizing: normalize leading '.'
-    suffix = f".{suffix.lstrip('.')}"
-
     name, psuffix = os.path.splitext(filename)
     if psuffix in SUPPORTED_FORMATS:
-        if suffix:
+        if suffix is not None:
+            suffix = normalize_extension_string(suffix)
+        if suffix in SUPPORTED_FORMATS and suffix != psuffix:
             mylog.warning(
-                "Ignoring supplied suffix '%s' in favour of '%s'", suffix, psuffix
+                "Received two valid image format '%s' and '%s'. "
+                "The former is ignored.",
+                psuffix,
+                suffix,
             )
+            return f"{name}{suffix}"
         return str(filename)
-    elif psuffix:
-        mylog.warning(
-            "Found unsupported suffix '%s', using '%s' instead.", psuffix, suffix
-        )
 
-    return f"{name}{suffix}"
+    if suffix is None:
+        suffix = ".png"
```

Comment:
Why not?
```suggestion
        suffix = ytcfg.get("yt", "default_image_format")
```

Reply:
I'm ambivalent. I don't like having a parameter with "default" in its name, makes it a bit hard to know what we're talking about. That being said I like the idea of parametrising a "preferred" output format (though that name isn't perfect either). I'd like to think about this a bit more and maybe do it in a separate PR.